ci-operator/step-registry/upi/conf/vsphere/upi-conf-vsphere-commands.sh

```diff
@@ -209,6 +209,23 @@ if test -f "${SHARED_DIR}/manifest_cluster-network-03-config.yml"
     cp "${SHARED_DIR}/manifest_cluster-network-03-config.yml" manifests/cluster-network-03-config.yml
 fi
 
+### Check External ccm setting
+if [ X"${CCM_TYPE}" != X"" ]; then
+  echo "set ccm type features ... "
+  platform_type=$(yq eval '.spec.platformSpec.type' manifests/cluster-infrastructure-02-config.yml)
+  lowplatform_type=$(echo $platform_type | tr '[:upper:]' '[:lower:]')
+  echo "platform type:  ${platform_type} ... ${lowplatform_type} ..."
+  pushd manifests
+  cat cluster-infrastructure-02-config.yml
+  yq eval -i 'del(.spec.platformSpec.'${lowplatform_type}')' cluster-infrastructure-02-config.yml
+  yq eval -i 'del(.spec.cloudConfig.key)' cluster-infrastructure-02-config.yml
+  yq eval -i 'del(.status.cpuPartitioning)' cluster-infrastructure-02-config.yml
+  yq eval -i 'del(.status.platformStatus.'${lowplatform_type}')' cluster-infrastructure-02-config.yml
+  yq eval ".spec.platformSpec.type = \"External\" | .spec.cloudConfig.name = \"""\" | .status.platform = \"External\" | .status.platformStatus.type = \"External\" | .status.platformStatus.external.cloudControllerManager.state = \"${CCM_TYPE}\"" -i cluster-infrastructure-02-config.yml
```

Comment:
Suggest to split the lines to multiple lines for easy reading.

Reply:
`/bin/bash: line 216: yq: command not found`
There is no yp either, I'll change this to use python.